src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/OAuth2/TokenResponse.cs

```diff
@@ -119,33 +116,47 @@ internal class TokenResponse
 
         internal static TokenResponse CreateFromBrokerResponse(IDictionary<string, string> responseDictionary)
         {
+            TokenResponse tokenResponse;
+
             if (responseDictionary.ContainsKey(TokenResponseClaim.ErrorDescription))
             {
-                return new TokenResponse
+                tokenResponse = new TokenResponse
                 {
-                    Error = responseDictionary[TokenResponseClaim.Error],
                     ErrorDescription = responseDictionary[TokenResponseClaim.ErrorDescription]
                 };
+                if (responseDictionary.ContainsKey(TokenResponseClaim.Error))
+                {
+                    tokenResponse.Error = responseDictionary[TokenResponseClaim.Error];
+                }
+                else
+                {
+                    // error_domain is a required field in a failed iOS broker response
+                    tokenResponse.Error = responseDictionary[TokenResponseClaim.ErrorDomain];
```

Comment:
Does MSAL have this logic?

Reply:
not sure...will check.

---
In reply to: [292153534](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1614#discussion_r292153534) [](ancestors = 292153534)